src/clj/puppetlabs/http/client/async.clj

```diff
@@ -119,9 +121,38 @@
       (.setDecompressBody (clojure.core/get opts :decompress-body true))
       (.setHeaders (:headers opts))))
 
+(schema/defn get-mean :- schema/Num
+  [timer :- Timer]
+  (->> timer
+       .getSnapshot
+       .getMean
+       (.toMillis TimeUnit/NANOSECONDS)))
+
+(defn get-metric-data
+  [timer]
+  (let [count (.getCount timer)
+        mean (get-mean timer)
+        aggregate (* count mean)]
+    {:count count
+     :mean mean
+     :aggregate aggregate}))
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
+(schema/defn get-client-metrics :- (schema/maybe common/Metrics)
+  "Returns the http client-specific metrics from the metric registry."
+  [metric-registry :- common/OptionalMetricRegistry]
+  (when metric-registry
+    (into {} (.getTimers metric-registry (ClientMetricFilter.)))))
+
+(schema/defn get-client-metrics-data :- (schema/maybe common/MetricsData)
+  "Returns a map of metric-id to metric data summary."
+  [metric-registry :- common/OptionalMetricRegistry]
+  (let [timers (get-client-metrics metric-registry)]
+    (into {} (map (fn [[metric-id timer]]
+                    {metric-id (assoc (get-metric-data timer) :metric-id metric-id)})
+                  timers))))
```

Comment:
nit: someday I need to write a little benchmark for this, but it seems like `reduce` would be faster since it doesn't have to iterate over the list twice.  I also think it's more readable but apparently I'm the only person who does :)